src/backend/providers/playnite/PlayniteProvider.cpp

```diff
@@ -0,0 +1,306 @@
+// Pegasus Frontend
+// Copyright (C) 2017-2020  M�ty�s Mustoha
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include "PlayniteProvider.h"
+
+#include "Log.h"
+#include "Paths.h"
+#include "PlayniteComponents.h"
+#include "PlayniteLaunchHelper.h"
+#include "PlayniteMetadataParser.h"
+#include "model/gaming/Assets.h"
+#include "model/gaming/Game.h"
+#include "providers/SearchContext.h"
+#include "utils/PathTools.h"
+
+#include <QTextDocument>
+
+namespace {
+QString default_installation()
+{
+    return paths::homePath() + QStringLiteral("/AppData/Roaming/Playnite/");
+}
+} // namespace
+
+namespace providers {
+namespace playnite {
+
+namespace {
+model::Game* create_file_path_game(const QString& path, model::Collection& collection, SearchContext& sctx)
+{
+    auto game_ptr = sctx.game_by_filepath(path);
+    if (!game_ptr) {
+        game_ptr = sctx.create_game_for(collection);
+        sctx.game_add_filepath(*game_ptr, path);
+    }
+    return game_ptr;
+}
+
+model::Game* create_uri_game(const QString& key, model::Collection& collection, SearchContext& sctx)
+{
+    auto game_ptr = sctx.game_by_uri(key);
+    if (!game_ptr) {
+        game_ptr = sctx.create_game_for(collection);
+        sctx.game_add_uri(*game_ptr, key);
+    }
+    return game_ptr;
+}
+
+void apply_game_fields(
+    model::Game& game,
+    const PlayniteComponents& components,
+    const PlayniteGame& game_info,
+    const QDir& playnite_dir)
+{
+    game.setTitle(game_info.name);
+
+    QTextDocument doc;
+    doc.setHtml(game_info.description);
+    const QString& description = doc.toPlainText();
```

Comment:
https://doc.qt.io/qt-5/qtextdocument.html#toPlainText returns a newly allocated QString, no need for the reference here.